rclpy/src/rclpy/_rclpy.c

```diff
@@ -1588,6 +1588,48 @@ rclpy_wait_set_add_entity(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Destroy the waitset structure
+/**
+ * \param[in] pywait_set Capsule pointing to the waitset structure
+ * \return NULL
+ */
+static PyObject *
+rclpy_destroy_wait_set(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pywait_set;
+
+  if (!PyArg_ParseTuple(args, "O", &pywait_set)) {
+    return NULL;
+  }
+  void * p = PyCapsule_GetPointer(pywait_set, NULL);
+  if (p == Py_None) {
+    PyErr_Format(PyExc_RuntimeError, "wait set is None");
```

Comment:
Same here that should result in the same error in `rcl_wait_set_fini`, not sure it needs to be checked in every layer

Reply:
Checking for Python `None` here is different from checking for `NULL` inside `rcl_wait_set_fini` and is therefore necessary.

Reply:
:+1: